src/platform/lumin-runtime/mxs-base-app.js

```diff
@@ -68,17 +68,34 @@ export class MxsBaseApp {
     this._prismControllers.push(controller);
 
     executor.callNativeAction(prism, 'setPrismController', controller);
+
+    const onDestroyHandlerId = executor.callNativeFunction(prism, 'onDestroyEventSub', (closingPrism) => {
+      const prismId = executor.callNativeFunction(closingPrism, 'getPrismId');
+      this._prisms = this._prisms.filter(p => executor.callNativeFunction(p, 'getPrismId') !== prismId);
+
+      if (!executor.callNativeFunction(closingPrism, 'onDestroyEventUnsub', onDestroyHandlerId)) {
```

Comment:
I think you should add a comment explaining that this is a workaround for the CallbackList deletion bug Wyck mentioned. Without context, It seems really strange that we have to unsubscribe from the (one-time-only) prism close event when we receive the prism close event (which will never fire again).

Reply:
Actually, are we totally sure this part is needed?

Reply:
Yes, we need to identify somehow that this prism doesn't exist anymore so `removePrism` doesn't try to work with it, otherwise the app crashes since MxsBaseApp attempts to delete it.

Reply:
I meant the onDestroyEventUnsub call, specifically, not removing it from the _prisms list.